plugins/worldinterface/src/worldproxy.cpp

```diff
@@ -907,23 +914,47 @@ GazeboYarpPlugins::Pose WorldProxy::getPose(const std::string& id)
       return ret;
     }
 
+    YarpWorldPose relative_link_pose;
+    if (frame_name!="")
+    {
+        physics::LinkPtr relative_link = HELPER_getLink(frame_name);
+        if( !relative_link )
+        {
+            yError() << "Unable to find specified link";
+            return ret;
+        }
+        #if GAZEBO_MAJOR_VERSION >= 8
+        relative_link_pose = relative_link->WorldPose();
+        #else
+        relative_link_pose = relative_link->GetWorldPose();
+        #endif
+    }
+
+    #if GAZEBO_MAJOR_VERSION >= 8
+    YarpWorldPose p=model->WorldPose();
+    YarpWorldPose final_pose (p.Pos()[0], p.Pos()[1], p.Pos()[2], p.Rot().Roll(), p.Rot().Pitch(), Rot().Yaw());
```

Comment:
`Rot().Yaw()`  should be replaced with  `p.Rot().Yaw()`, right?

Reply:
Done! see 56ef8787ebd357a395ecc822e90acf8c20dd99b9
Sorry for the delay!